### PR TITLE
[WIP] Harmony dev

### PR DESCRIPTION
**Pull request by yzhoujuly**
## Description
Adding Harmony to TrustWallet. Up coming commit will remove some files from this PR

## Testing instructions
Using bootstrap.sh to build 

## Types of changes
Adding Harmony to TrustWallet

## Checklist

- [X] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.

### CODE REVIEW

File: coins.json [hunk 1]
```diff
@@ -1,4 +1,22 @@
 [
+    {
+        "id": "harmony",
+        "name": "Harmony",
+        "symbol": "ONE",
+        "decimals": 18,
+        "blockchain": "Harmony",
+        "derivationPath": "m/44'/60'/0'/0/0",
```
Comment (vikmeup) [inline coins.json]: wrong derivation path, 60 is ethereum

File: src/Coin.cpp [hunk 2]
```diff
@@ -92,6 +95,9 @@ bool TW::validateAddress(TWCoinType coin, const std::string& string) {
         return Bitcoin::SegwitAddress::isValid(string, hrp) ||
                Groestlcoin::Address::isValid(string, {p2pkh, p2sh});
 
+    // YZ:
```
Comment (vikmeup) [inline src/Coin.cpp]: why put `YZ` comment?

File: tests/Harmony/SignerTests.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                                       /* gasPrice: */ 20000000000,
+                                       /* gasLimit: */ 21000,
+                                       /* to: */ address,
+                                       ///* amount: */ 0x1158e460913d00000,
+                                       //Err: integer literal is too large to be represented in any integer type
+                                       
+                                       /* amount: */ 0x58e460913d,
+                                       
+                                       /* payload: */ {}
+                                       );
```
Comment (coolcottontail) [inline tests/Harmony/SignerTests.cpp]: please update using harmony's transaction.

File: tests/interface/TWCoinTypeConfigTests.cpp [hunk 4]
```diff
@@ -147,6 +153,10 @@ TEST(TWCoinTypeConfiguration, TWCoinTypeConfigurationGetSymbol) {
 }
 
 TEST(TWCoinTypeConfiguration, TWCoinTypeConfigurationGetDecimals) {
+    
+    // YZ
+    ASSERT_EQ(TWCoinTypeConfigurationGetDecimals(TWCoinTypeHarmony), 18);
```
Comment (coolcottontail) [inline tests/interface/TWCoinTypeConfigTests.cpp]: 18 is same value as Ethereum?

File: tests/interface/TWHarmonySignerTests.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    // Check proto storing
+    Ethereum::Proto::SigningInput input;
+    auto storedData = store(value);
+    input.set_gas_limit(storedData.data(), storedData.size());
+    ASSERT_EQ(hex(input.gas_limit()), hex(expectedData));
+
+    // Check proto loading
+    auto protoLoaded = load(input.gas_limit());
+    ASSERT_EQ(protoLoaded, value);
+}
```
Comment (coolcottontail) [inline tests/interface/TWHarmonySignerTests.cpp]: is there any test for signing ? as the name of this file is SignerTest.cpp  ?

**Comment by vikmeup** [comment]
Closing for now, until implemented - re open then. Create PR as Draft if still work in progress

**Pull request closed without merging**